tests/lib/EventDispatcher/EventListener/ContentSuggestionSubscriberTest.php

```diff
@@ -8,67 +8,161 @@
+    {
+        $this->configProviderMock->method('getRepositoryConfig')
+            ->willReturn(['search' => ['engine' => 'not_legacy']]);
+        $this->searchServiceSupportsScoring = false;
+        $this->performTestOnContentSuggestion();
+    }
+
+    public function testOnContentSuggestionWithException(): void
+    {
+        $this->configProviderMock->method('getRepositoryConfig')
+            ->willReturn(['search' => ['engine' => 'legacy']]);
+        $this->searchServiceSupportsScoring = true;
+
+        $query = new SuggestionQuery('test', 10, 'eng-GB');
+        $searchService = $this->getSearchServiceMockWithException();
+        $mapper = $this->getSearchHitToContentSuggestionMapperMock();
+
+        $subscriber = new ContentSuggestionSubscriber($this->configProviderMock, $searchService, $mapper);
+        $subscriber->setLogger($this->loggerMock);
+
+        $event = new BuildSuggestionCollectionEvent($query);
+
+        $this->loggerMock->expects($this->once())->method('error');
+
+        $subscriber->onBuildSuggestionCollectionEvent($event);
+    }
+
+    private function performTestOnContentSuggestion(): void
     {
         $query = new SuggestionQuery('test', 10, 'eng-GB');
         $searchService = $this->getSearchServiceMock();
         $mapper = $this->getSearchHitToContentSuggestionMapperMock();
 
-        $subscriber = new ContentSuggestionSubscriber($searchService, $mapper);
+        $subscriber = new ContentSuggestionSubscriber($this->configProviderMock, $searchService, $mapper);
 
         $event = new BuildSuggestionCollectionEvent($query);
         $subscriber->onBuildSuggestionCollectionEvent($event);
 
         $collection = $event->getSuggestionCollection();
 
         self::assertCount(1, $collection);
+
+        self::assertNotNull($this->capturedQuery);
+        self::assertNotEmpty($this->capturedQuery->sortClauses);
     }
 
-    /**
-     * @return \PHPUnit\Framework\MockObject\MockObject|\Ibexa\Core\Repository\SiteAccessAware\SearchService
-     */
     private function getSearchServiceMock(): SearchService
     {
         $searchServiceMock = $this->createMock(SearchService::class);
-        $searchServiceMock->method('findContent')->willReturn(
-            new SearchResult(
-                [
-                    'searchHits' => [
-                        $this->createMock(SearchHit::class),
-                    ],
-                ]
-            )
-        );
+        $searchServiceMock->method('supports')
+            ->willReturnCallback(function ($capability) {
+                return $capability === SearchServiceInterface::CAPABILITY_SCORING && $this->searchServiceSupportsScoring;
+            });
+        $searchServiceMock->method('findContent')->willReturnCallback(function ($query) {
+            $this->capturedQuery = $query;
+
+            return new SearchResult(['searchHits' => [$this->createMock(SearchHit::class)]]);
+        });
```

Comment:
```suggestion
        $searchServiceMock->method('findContent')
            ->willReturnCallback(function ($query) {
                $this->capturedQuery = $query;

                return new SearchResult(['searchHits' => [$this->createMock(SearchHit::class)]]);
            });
```
formatting like above